addons/point_of_sale/static/src/js/screens.js

```diff
@@ -137,15 +142,29 @@ var ScreenWidget = PosBaseWidget.extend({
             this.$el.removeClass('oe_hidden');
         }
 
-        this.pos.barcode_reader.set_action_callback({
-            'cashier': _.bind(self.barcode_cashier_action, self),
-            'product': _.bind(self.barcode_product_action, self),
-            'weight': _.bind(self.barcode_product_action, self),
-            'price': _.bind(self.barcode_product_action, self),
-            'client' : _.bind(self.barcode_client_action, self),
-            'discount': _.bind(self.barcode_discount_action, self),
-            'error'   : _.bind(self.barcode_error_action, self),
-        });
+        if (this.$el.first().hasClass("receipt-screen")){
+            this.pos.barcode_reader.reset_action_callbacks();
+        } else if (this.$el.first().hasClass("payment-screen")){
```

Comment:
Cant we override the code in the right class and move set_action_callback logic into a functino

Reply:
move the code to relevant class.

Reply:
And maybe ignore instead of error.

Reply:
@antonylesuisse I think it's safer to do this suggested refactoring in master?